### DISCUSSION TITLE
Add  "orderbyclause" as a Select property

### DISCUSSION THREAD

**Discussion by dhait** [sqlalchemy/sqlalchemy#10290] [Ideas]
The Select object has a property called `whereclause`, which allows you to extract the "WHERE" clause from the query.  There should be an equivalent property called `orderbyclause` to get the portion of the query which is "ORDER BY".

The Select object already has a private member called `_order_by_clause`, so it shouldn't be too hard to make this public.

This would be useful for certain use cases, such as building up a select query from a bunch of independent pieces (see for example, [https://stackoverflow.com/questions/76970534/sqlalchemy-2-0-query-builder-in-model-like-in-rails-activerecord](https://stackoverflow.com/questions/76970534/sqlalchemy-2-0-query-builder-in-model-like-in-rails-activerecord)

**Comment by zzzeek** [comment]
hmm looks like a public `.whereclause` sneaked in there at some point, so that sets up a precedent for `order_by_clause` also (I know `whereclause` is one word, but that seems a bit historical here, I think underscores would be good for this name). 

I looks like `.whereclause` is returning a standard `ClauseList` kind of structure, so we're exposing that OK, `order_by_clause` can also expose as a `ClauseList`   (for `whereclause`, the "separator" for `ClauseList` is "AND".  For `order_by_clause`, the separator is a comma.   makes sense?)

**Reply by CaselIT** [reply]
seems kinda arbitrary to have only (?) where clause and order by. But it should be easy to just add a prop there

**Reply by zzzeek** [reply]
I prefer to look it as that we are giving each new attribute the 5 years of celebratory introduction it needs
